src/core/lib/security/authorization/rbac_policy.cc

```diff
@@ -200,8 +202,12 @@ Rbac::Principal::Principal(Principal::RuleType type, Principal principal)
 }
 Rbac::Principal::Principal(Principal::RuleType type) : type(type) {}
 Rbac::Principal::Principal(Principal::RuleType type,
-                           StringMatcher string_matcher)
-    : type(type), string_matcher(std::move(string_matcher)) {}
+                           absl::optional<StringMatcher> string_matcher)
+    : type(type), string_matcher(std::move(string_matcher)) {
+  if (type == RuleType::kPath) {
```

Comment:
This is another good example of why it would be a good idea to have a separate factory method for each rule type, as I suggested in your other PR.  You can have the factory method for the "path" rule type take `StringMatcher` instead of `absl::optional<StringMatcher>`, so the API enforces that the matcher is always specified for that rule type, and you don't need a separate assertion here.

Reply:
Yeah, I agree. Thanks for explaining:) I updated the other PR. Once that PR is merged, I can make the absl::optional related updates in this PR.

Reply:
@markdroth The PR is ready for review. PTAL. Thank you!:)